### ISSUE TITLE
[GlobalStep] "More" block on Pages/Post and "Read More" block on product description fails to work on frontend

### ISSUE THREAD

**Issue by gglobalstep** [woocommerce/woocommerce#29427]
## Bug Description:
"More" block on Pages/Post and "Read More" block on product description fails to work on the frontend

## Test Instructions:
Critical Flows-Add Simple Product to Cart
https://github.com/woocommerce/woocommerce/wiki/Critical-Flows

## Environment:
Woocommerce Version: WooCommerce 5.1 and 5.0
WordPress version: (v5.7)
WooCommerce Admin version: v2.0.2

PC:
Windows 10, Mac 10.14.6
Chrome(Version 88.0.4324.104) 
Firefox(Version 86.0)
Safari: v13.1.1

## Steps To Reproduce:
1. Create any test site using JN site.
1. Install and activate all the required plugins
1. Complete the onboarding setup wizard.
1. Go to Products->Add New
1. Under Product Description, add content
1. Insert a "read more" block in between the content 
1. Click on "Publish" button and then "Preview" button
1. Note that  "Read More" block on product description fails to work on frontend
1. Backend: Go to Pages-Add New or Post->Add New 
1. Add multiple paragraph blocks with content
1. Add a "More" block in between the paragraphs
1. Click on "Publish" button and then "Preview" button
1. "More" block on Pages fails to work on frontend

## Actual Result:
"More" block on Pages/Post and "Read More" block on product description fails to work on the frontend

## Expected Result:
"More" block on Pages/Post and "Read More" block on product description should work on the frontend

## Video:

https://user-images.githubusercontent.com/41110392/111489654-56ee4c80-8760-11eb-8d10-68b951bf40d4.mov

**Isolating the problem (mark completed items with an [x]):**
- [ ] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [ ] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [x] I can reproduce this bug consistently using the steps above.

**Comment by juliaamosova** [comment]
Adding the `needs developer feedback` label here so that one of the Core developers could confirm expected behavior. 

**Tl;dr:** The Read More tag of the Classic WordPress Editor doesn't work in the Product Description. See the gif that shows my experience below:

http://g.recordit.co/4putp98Ypz.gif

Is it expected behavior or is it a bug?

**Comment by ObliviousHarmony** [comment]
After reviewing this with the team we have agreed that this is the expected behaviour, as per @nerrad's comment.
